biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -418,6 +430,253 @@ end subroutine get_harvest_rate_area
 
    ! ============================================================================
 
+   subroutine get_harvestable_carbon (csite, site_area, hlm_harvest_catnames, harvestable_forest_c )
+
+     !USES:
+     use SFParamsMod,  only : SF_val_cwd_frac
+     use EDTypesMod,   only : AREA_INV
+
+
+     ! -------------------------------------------------------------------------------------------
+     !
+     !  DESCRIPTION:
+     !  get the total carbon availale for harvest for three different harvest categories:
+     !  primary forest, secondary mature forest and secondary young forest
+     !  under two different scenarios:
+     !  harvestable carbon: aggregate all cohorts matching the dbhmin harvest criteria
+     !
+     !  this subroutine shall be called outside the patch loop
+     !  output will be used to estimate the area-based harvest rate (get_harvest_rate_carbon)
+     !  for each cohort.
+
+     ! Arguments
+     type(ed_site_type), intent(in), target :: csite
+     real(r8), intent(in) :: site_area    ! temporary variable
+     character(len=64), intent(in) :: hlm_harvest_catnames(:) ! names of hlm harvest categories
+
+     real(r8), intent(out) :: harvestable_forest_c(hlm_num_lu_harvest_cats)
+
+     ! Local Variables
+     type(ed_patch_type), pointer  :: currentPatch
+     type(ed_cohort_type), pointer :: currentCohort
+     real(r8) :: harvestable_patch_c     ! patch level total carbon available for harvest, kgC site-1
+     real(r8) :: harvestable_cohort_c    ! cohort level total carbon available for harvest, kgC site-1
+     real(r8) :: sapw_m      ! Biomass of sap wood
+     real(r8) :: struct_m    ! Biomass of structural organs
+     integer :: pft         ! Index of plant functional type
+     integer :: h_index     ! for looping over harvest categories
+
+     ! Initialization
+     harvestable_forest_c = 0._r8
+
+     ! loop over patches
+     currentPatch => csite%oldest_patch
+     do while (associated(currentPatch))
+        harvestable_patch_c = 0._r8
+        currentCohort => currentPatch%tallest
+
+        do while (associated(currentCohort))
+           pft = currentCohort%pft
+
+           ! only account for cohorts matching the following conditions
+           if(int(prt_params%woody(pft)) == 1)then ! only set logging rates for trees
+              sapw_m   = currentCohort%prt%GetState(sapw_organ, all_carbon_elements)
+              struct_m = currentCohort%prt%GetState(struct_organ, all_carbon_elements)
```

Comment:
@rgknox @sshu88 with the nutrients V2 update, the `all_carbon_elements` id option has been removed.  Should this be changed to the `carbon_elements_list` and placed in an element loop to sum up the carbon prior to calculating `harvestable_cohort_c` below?

For reference, the relevant commit that removed `all_carbon_elements` is https://github.com/NGEET/fates/commit/c4422f1e0781b46ea31ee8c206b97fffb179b9d1

Reply:
@glemieux Yes. The total biomass is the sum of C12 and any other isotopes if activated.

Reply:
I removed all_carbon_elements because we don't yet track mass flows of isotopes (we do have some isotope functions, but no mass tracking).  The use of all_carbon_elements was making a very low level function in parteh use some extra logic and looping.  carbon12_element has the same effect in all code since there are no other tracked isotopes, and is faster